sql/catalyst/src/main/scala/org/apache/spark/sql/internal/connector/SupportsPushDownCatalystFilters.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.internal.connector
+
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.sources.Filter
+
+/**
+ * A mix-in interface for {@link FileScanBuilder}. This can be used to push down filters to
+ * FileIndex in the format of catalyst Expression.
+ */
+trait SupportsPushDownCatalystFilters {
+  /**
+   * Pushes down filters to FileIndex in the format of catalyst Expression. The filters will be
+   * separated into partition filters and data filters. The data filters that are pushed to the
+   * data source and the date filters that need to be evaluated after scanning are returned.
+   */
+  def pushCatalystFilters(partitionFilters: Seq[Expression]): (Array[Filter], Seq[Expression])
```

Comment:
can we follow `SupportsPushDownFilters`?
```
trait SupportsPushDownCatalystFilters {
  def pushFilters(Seq[Expression] filters): Seq[Expression]
  def pushedFilters(): Seq[Expression]
}
```

Reply:
Done :)